lib/active_model_serializers/model.rb

```diff
@@ -67,5 +81,21 @@ def self.lookup_ancestors
       [self]
     end
     # :nocov:
+
+    module AttributesAreAlwaysTheInitializationData
+      def initialize(attributes = {})
+        @initialized_attributes = attributes && attributes.symbolize_keys
+        super
+      end
+
+      # Defaults to the downcased model name.
+      def id
+        @initialized_attributes.fetch(:id) { self.class.model_name.name && self.class.model_name.name.downcase }
+      end
+
+      def attributes
+        @initialized_attributes
```

Comment:
yup, they can be mutated https://github.com/rails-api/active_model_serializers/pull/1998/files#r93893962